index.ts

```diff
@@ -14,18 +14,18 @@
  * to cleanly shut down before the process exits.
  */
 
-// TODO(rictic): convert this library to promises.
-
 /**
  * The type of a cleankill interrupt handler.
  */
-export type Handler = (doneHandling: () => void) => void;
+export type Handler = () => Promise<void>;
 
-let interruptHandlers: Handler[] = [];
+const interruptHandlers: Handler[] = [];
 
 /**
- * Register a handler to occur on SIGINT. All handlers are passed a callback,
- * and the process will be terminated once all handlers complete.
+ * Register a handler to occur on SIGINT. The handler must return a promise if
```

Comment:
I'm just coming at it from the "make all one-time async signals Promises" point of view. Not that you'd want to do this, but if you're using async/await, you could:

```javascript
let interrupt = await cleankill.onInterrupt();
cleanupSync();
```

I just think you never know where you get benefits from single, composable ways of representing future state. No change needed here though.